### PR TITLE
prosody: 0.11.13 -> 0.12.0

### PR DESCRIPTION
**Pull request by picnoir**
It's been quite a journey...

Prosody 0.12 requires luaunbound, we package it on https://github.com/NixOS/nixpkgs/commit/811f95f000d9610bd85cca1cf78ae9a069464d87. See commit message for more details.  Cc @teto, you seem to be the one maintaining most of this packageset. 

The last Slixmpp bump revealed a upstream bug. Fixing it on https://github.com/NixOS/nixpkgs/commit/0e630462da8c2e83b6d04eb935353d1e34549c32 to unbreak the prosody test. See commit message for more details. Cc @fabaff, you seem to be relying on slixmpp as well.

We finally update prosody on https://github.com/NixOS/nixpkgs/commit/2585ca317fa51a0005b04cac01f70b1f68f71674. We remove the recently introduced wrappers in the process. I don't really get why they aren't necessary anymore. Cc @arcnmx

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by teto** [comment]
> however it did break a fair share of Lua packages.

do you mean on the lua side or  nix expressions ?

> Cc @teto, you seem to be the one maintaining most of this packageset.

sadly yes. Let me know if I can help improve stuff or if you wanna help. I have a few ideas already about what is wrong but having external feedback would be good to know.

File: pkgs/development/python-modules/slixmpp/default.nix [hunk 1]
```diff
@@ -39,6 +39,8 @@ buildPythonPackage rec {
       src = ./hardcode-gnupg-path.patch;
       inherit gnupg;
     })
+    # Upstream MR: https://lab.louiz.org/poezio/slixmpp/-/merge_requests/198
+    ./0001-xep_0030-allow-extra-args-in-get_info_from_domain.patch
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/slixmpp/default.nix:43]: Why not use fetchpatch?

File: pkgs/development/lua-modules/generated-packages.nix [hunk 2]
```diff
@@ -1944,6 +1944,31 @@ buildLuarocksPackage {
...
+    sha256 = "1lsh0ylp5xskygxl5qdv6mhkm1x8xp0vfd5prk5hxkr19jk5mr3d";
+  };
+
+  disabled = with lua; (luaOlder "5.1") || (luaAtLeast "5.5");
+  propagatedBuildInputs = [ lua ];
+
+  meta = {
+    homepage = "https://www.zash.se/luaunbound.html";
+    description = "A binding to libunbound";
+    license.fullName = "MIT";
```
Comment (SuperSandro2000) [inline pkgs/development/lua-modules/generated-packages.nix:1968]: Why is luamodules not using lib.licenses?

File: pkgs/development/python-modules/slixmpp/default.nix (hunk 1 above)
Comment (NinjaTrappeur) [reply pkgs/development/python-modules/slixmpp/default.nix:43]: Read the commit message.

Comment (SuperSandro2000) [reply pkgs/development/python-modules/slixmpp/default.nix:43]: Please put that in the comment or filter the files with fetchpatch.

**Comment by Izorkin** [comment]
@NinjaTrappeur modules bookmarks and smack are already included in the main package:
```
l ./result/lib/prosody/modules | egrep 'mod_bookmarks|smacks'
dr-xr-xr-x  2 root root 4.0K Jan  1  1970 mod_bookmarks
-r--r--r--  1 root root  17K Jan  1  1970 mod_bookmarks.lua
dr-xr-xr-x  2 root root 4.0K Jan  1  1970 mod_smacks
-r--r--r--  1 root root  29K Jan  1  1970 mod_smacks.lua
```

```diff
diff --git a/pkgs/servers/xmpp/prosody/default.nix b/pkgs/servers/xmpp/prosody/default.nix
index 194c89879f9..0d00e48f48a 100644
--- a/pkgs/servers/xmpp/prosody/default.nix
+++ b/pkgs/servers/xmpp/prosody/default.nix
@@ -28,10 +28,8 @@ stdenv.mkDerivation rec {
   # prosody module to comply with XEP-0423 and provide a working
   # default setup.
   nixosModuleDeps = [
-    "bookmarks"
     "cloud_notify"
     "vcard_muc"
-    "smacks"
     "http_upload"
   ];
   src = fetchurl {

```

**Comment by NinjaTrappeur** [comment]
Nice catch @Izorkin, PTAL

**Comment by NinjaTrappeur** [comment]
@teto 

> do you mean on the lua side or nix expressions ?

On the Lua side.

> sadly yes. Let me know if I can help improve stuff or if you wanna help. I have a few ideas already about what is wrong but having external feedback would be good to know.

I pinged you mostly to make sure I did not break your setup :) I sadly do not have much experience nor interest in Lua.

I only use of it through my Prosody instance, which has been a constant pain to maintain across the years. Especially on NixOS: this software makes a lot of filesystem location assumptions :(

Thanks for your hard work, it greatly simplified this Prosody bump. Kudos <3

**Comment by yu-re-ka** [comment]
This seems to work fine for me. Can we merge it?

File: pkgs/development/lua-modules/generated-packages.nix (hunk 2 above)
Comment (teto) [reply pkgs/development/lua-modules/generated-packages.nix:1968]: licences are copied from the rockspec which uses freeform strings

**Comment by teto** [comment]
> It's been quite a journey...

I know the feeling so I merge as is. I've noticed that prosody has some busted tests. It would be nice if we could enable these in the future. I want to experiment with luarocks "flat" installs (i.e., dont install in a subdirectory lib/lua5.2 but at the root of the derivation directly) and to minimize breakage/help with dev, I try to enable tests on several packages. 
This may also prevent breakage when updating the luarocks package set. Feel free to ask me should you have any question on the matter.

**Comment by NinjaTrappeur** [comment]
For the record, @ju1m wrote a patch that introduce the `mod_http_file_share` support in the NixOS module there: https://github.com/NinjaTrappeur/nixpkgs/pull/1/commits/beb292e3ff3080ae73a3f2c647cd72bf67ceea6f

@ju1m, you probably could open a new PR against Nixpkgs now we have this PR on master.

**Pull request merged**